onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -653,6 +681,170 @@ Status ReduceKernel<true>::ComputeImpl<int32_t, CUDNN_REDUCE_TENSOR_NO_INDICES>(
   return Status::OK();
 }
 
+template <>
+template <>
+Status ReduceKernel<true>::ComputeImpl<int8_t, CUDNN_REDUCE_TENSOR_NO_INDICES>(OpKernelContext* ctx, cudnnReduceTensorOp_t cudnn_reduce_op) const {
+  typedef typename ToCudaType<int8_t>::MappedType CudaT;
```

Comment:
Is there something significantly different about the int8_t and uint8_t implementations or could we use the same code for each and just not specialize on T?

Reply:
Yes, int32, int8 and uint8 are similar but significantly different from the float/double.

---
In reply to: [400585373](https://github.com/microsoft/onnxruntime/pull/3341#discussion_r400585373) [](ancestors = 400585373)